src/conv/invokers/impl_gemm_dynamic.cpp

```diff
@@ -20,53 +20,175 @@ float CallImplicitGemmDynamic(const miopen::Handle& handle,
 
     auto kernel = kernels[0];
     MIOPEN_LOG_I(kernel.GetName());
-    bool kernel_is_1x1 = (kernel.GetName().find("igemm_v4r1_1x1_dynamic") == 0);
-    // clang-format off
-    int hi          = ctx.in_height;
-    int wi          = ctx.in_width;
-    int n           = ctx.batch_sz;
-    int k           = ctx.n_outputs;
-    int c           = ctx.n_inputs;
-    int ho          = ctx.out_height;
-    int wo          = ctx.out_width;
-    int stride_h    = ctx.kernel_stride_h;
-    int stride_w    = ctx.kernel_stride_w;
-    int dilation_h  = ctx.kernel_dilation_h;
-    int dilation_w  = ctx.kernel_dilation_w;
-    int pad_h       = ctx.pad_h;
-    int pad_w       = ctx.pad_w;
-    int y           = ctx.kernel_size_h;
-    int x           = ctx.kernel_size_w;
-    int __pack0     = 0;
-    // clang-format on
-    std::vector<OpKernelArg> opArgs;
-    opArgs.emplace_back(src);
-    opArgs.emplace_back(wei);
-    opArgs.emplace_back(dst);
-    opArgs.emplace_back(hi);
-    opArgs.emplace_back(wi);
-    opArgs.emplace_back(n);
-    opArgs.emplace_back(k);
-    opArgs.emplace_back(c);
-    opArgs.emplace_back(ho);
-    opArgs.emplace_back(wo);
-    opArgs.emplace_back(stride_h);
-    opArgs.emplace_back(stride_w);
-    opArgs.emplace_back(dilation_h);
-    opArgs.emplace_back(dilation_w);
-    opArgs.emplace_back(pad_h);
-    opArgs.emplace_back(pad_w);
-    if(kernel_is_1x1)
+    if(ctx.direction.IsForward())
     {
-        opArgs.emplace_back(__pack0);
+        bool kernel_is_1x1 = (kernel.GetName().find("igemm_v4r1_1x1_dynamic") == 0);
```

Comment:
Please use ConvolutionContextInterpreter https://github.com/ROCmSoftwarePlatform/MIOpen/blob/16d87c1952e41b54b36fe6cb59a2a40cc3275934/src/solver/implicitgemm_util.hpp#L100
that gives hi/wi/c depending upon the forward/backward direction. You could simply invoke that code before if else block.

Reply:
let me refactor code here

Reply:
@carlushuang Please raise an another PR to do that thing. As of now, this piece of code could be left as it is.